airflow/providers/google/cloud/hooks/cloud_batch.py

```diff
@@ -0,0 +1,229 @@
+        )
+
+        tasks: pagers.ListTasksPager = self.get_conn().list_tasks(
+            request=list_tasks_request)
+
+        return self._limit_list(tasks, limit)
+
+    def wait_for_job(
+            self,
+            job_name: str,
+            polling_period_seconds: float = 10,
+            timeout: Union[float, None] = None
+    ) -> Job:
+        client = self.get_conn()
+        while timeout is None or timeout > 0:
+            try:
+                job = client.get_job(name=f"{job_name}")
+                status: JobStatus.State = job.status.state
+                if status == JobStatus.State.SUCCEEDED \
+                        or status == JobStatus.State.FAILED \
+                        or status == JobStatus.State.DELETION_IN_PROGRESS:
+                    return job
+                else:
+                    sleep(polling_period_seconds)
+            except Exception as e:
+                self.log.exception(
+                    "Exception occurred while checking for job completion.")
+                raise e
+
+            if timeout is not None:
+                timeout -= polling_period_seconds
+
+        raise AirflowException(f"Job with name [{job_name}] timed out")
+
+    def get_job(self, job_name) -> Job:
+        return self.get_conn().get_job(name=f"{job_name}")
+
+    def _limit_list(self, jobs, limit):
+        result = []
+        for item in jobs:
+            if limit is not None and limit == 0:
+                break
+            result.append(item)
+            if limit is not None:
+                limit -= 1
+        return result
+
+
+class CloudBatchAsyncHook(GoogleBaseHook):
+    """
+    Async hook for the Google Cloud Batch service.
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account.
+    """
+    
+    def __init__(
+        self,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ):
+        if kwargs.get("delegate_to") is not None:
+            raise RuntimeError(
+                "The `delegate_to` parameter has been deprecated before and finally removed in this version"
+                " of Google Provider. You MUST convert it to `impersonate_chain`"
+            )
+
+        self._client: BatchServiceAsyncClient = BatchServiceAsyncClient()
```

Comment:
```suggestion
        self._client: BatchServiceAsyncClient | None = None
```
We leave the client instantiation out of the constructor.